src/table.rs

```diff
@@ -236,6 +238,77 @@ impl Table {
         self.rows.len()
     }
 
+    /// Sorts the rows of the table using a custom comparator function.
+    ///
+    /// The sorting is done in place and modifies the original table.
+    ///
+    /// ## Example
+    ///
+    /// The following example demonstrates how to sort the rows of a table
+    /// in descending order based on the content of the first column:
+    ///
+    /// ```
+    /// // Create a new table with a header and several rows
+    /// let mut table = Table::new();
+    /// table
+    ///     .set_header(vec![Cell::new("Names")])
+    ///     .add_row(vec![Cell::new("Alexander the Great")])
+    ///     .add_row(vec![Cell::new("Shahd Aldahar")])
+    ///     .add_row(vec![Cell::new("Neil Armstrong")])
+    ///     .add_row(vec![Cell::new("Hassan Ibn Al-haitham")])
+    ///     .add_row(vec![Cell::new("Al-khwarizmi")]);
+    ///
+    /// // Sort the rows in descending order based on the first column
+    /// table.sort_rows(|a, b| b.cells()[0].cmp(&a.cells()[0]));
+    ///
+    /// // Print the table after sorting
+    /// println!("Sort in descending order:");
+    /// println!("{table}");
+    /// ```
+    ///
+    /// **Output:**
+    /// ```plaintext
+    /// Sort in descending order:
+    /// +-----------------------+
+    /// | Names                 |
+    /// +=======================+
+    /// | Shahd Aldahar         |
+    /// |-----------------------|
+    /// | Neil Armstrong        |
+    /// |-----------------------|
+    /// | Hassan Ibn Al-haitham |
+    /// |-----------------------|
+    /// | Alexander the Great   |
+    /// |-----------------------|
+    /// | Al-khwarizmi          |
+    /// +-----------------------+
+    /// ```
+    ///
+    /// ## Arguments
+    ///
+    /// * `compare` - A comparator function that defines the sorting logic.
+    ///   The function should take two rows (`&Row`) and return an `Ordering`
+    ///   (`Ordering::Less`, `Ordering::Equal`, or `Ordering::Greater`).
+    ///
+    /// ## Notes
+    ///
+    /// - The sorting is done based on the provided comparison function.
+    /// - If you need to sort by multiple columns, you can implement more complex
+    ///   logic inside the comparator function.
+    /// 
+    /// ## Time Complexity
+    ///
+    /// The time complexity of this function is `O(n log n)`, where `n` is the number
+    /// of rows in the table. This is due to the use of the `sort_by` method, which
+    /// typically implements an efficient sorting algorithm such as Timsort or
+    /// Quicksort.
```

Comment:
Don't try to be smart about this stuff.
If you want to include info about complexity just say that you're using the std lib sorting function. Ngl, this feels a bit like AI generated chatter :D 

In the end, this won't be problematic for most people, as it's a fast algorithm and people won't create tables with thousands of entries. I would just discard this section.

Reply:
I found this information in the standard documentation. So, I added it, thinking it might be useful, but perhaps nobody cares.